infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespace.java

```diff
@@ -153,10 +153,10 @@ void prepare(boolean canCreate, Map<String, String> labels, Map<String, String>
         throw new InfrastructureException(
             format("Creating the namespace '%s' is not allowed, yet it was not found.", name));
       }
-      namespace = create(name, client);
+      create(name, client);
     }
-    label(namespace, labels);
-    annotate(namespace, annotations);
+    label(client.namespaces().withName(name).get(), labels);
+    annotate(client.namespaces().withName(name).get(), annotations);
```

Comment:
is it just refactoring? I'm not sure which benefits it provides e.g. `client.namespaces().withName(name).get()` is called twice. I like the original version more

Reply:
To label and to annotation, we need a new fresh namespace object.
Otherwise adding annotation removes previously added labels.

Reply:
sorry, for not following 
I looked at the label method and it seems to merge the existing and new labels - https://github.com/eclipse-che/che-server/pull/355/files#diff-2e663cdd99b84ca2ecf99c9d2df6edc6f1036866f87a175a2a7d8ae05f9aad7bR196

Could you clarify how this change is relevant to the PR?